Assets/__Scripts/BeatSaberSong.cs

```diff
@@ -220,6 +221,30 @@ public JSONNode ToJSONNode()
         }
     }
 
+    public DateTime LastWriteTime;
+    private bool _isFavourite;
+    public bool IsFavourite
+    {
+        get => _isFavourite;
+        set
+        {
+            var path = Path.Combine(directory, ".favourite");
```

Comment:
May I ask why we're using the filesystem for tracking favorites? If cross-editor compatibility is not an issue then we can also consider utilizing the `editors` object and store it there.

Reply:
Using the editors option was kinda second on the list.
The main reason I didn't do that is that currently we always save the diffs as well as the Info.dat and on bigger maps that's really slow. It would also mean that maps downloaded off beatsaver would randomly be favourited while the hidden file should not get packaged. And, as you pointed out, is is something that MMA2 could use too.